packages/office-ui-fabric-react/src/components/Checkbox/Checkbox.tsx

```diff
@@ -46,7 +46,7 @@ export class Checkbox extends BaseComponent<ICheckboxProps, ICheckboxState> impl
 
     const { isFocused } = this.state;
     const isChecked = checked === undefined ? this.state.isChecked : checked;
-
+    const inputName = (inputProps && inputProps.name) ? inputProps.name : this._id;
```

Comment:
For the individual checkbox's the props should be changed to extend  React.HTMLattributes<HTMLInputElement> (I think), that way, again, name is a standard prop and it more accurately reflects what a checkbox is, a type of input element.